### PR TITLE
Fix covariance serialisation

### PR DESCRIPTION
**Pull request by AtreyeeS**
This fixes the issue with serialising compound models as discussed in #4124

I also added a serialisation example in the fermi lat tutorial

### CODE REVIEW

File: examples/tutorials/data/fermi_lat.py [hunk 1]
```diff
@@ -229,14 +229,21 @@
 # we just load one that represents a small cutout for the Galactic center
 # region.
 #
+# For serialisation of TemplateSpatialModels, it is always useful to specify
+# the filename.
+#
 
 diffuse_galactic_fermi = Map.read("$GAMMAPY_DATA/fermi-3fhl-gc/gll_iem_v06_gc.fits.gz")
```
Comment (registerrier) [inline examples/tutorials/data/fermi_lat.py:236]: This is no logner needed

File: gammapy/modeling/models/tests/test_io.py [hunk 2]
```diff
@@ -362,3 +366,22 @@ def test_to_dict_not_default():
...
+
+
+def test_compound_models_io(tmp_path):
+    m1 = PowerLawSpectralModel()
+    m2 = LogParabolaSpectralModel()
+    m = CompoundSpectralModel(m1, m2, operator.add)
+    sk = SkyModel(spectral_model=m, name="model")
+    Models([sk]).write(tmp_path / "test.yaml")
+    sk1 = Models.read(tmp_path / "test.yaml")
+    assert_allclose(sk1.covariance.data, sk.covariance.data, rtol=1e-3)
```
Comment (registerrier) [inline gammapy/modeling/models/tests/test_io.py:378]: Here the covariance should be equal to 0 no?

**Review by registerrier — DISMISSED** [review]
Thanks @AtreyeeS .  Looks good. I have left some comments inline

**Review by adonath — APPROVED** [review]
Thanks @AtreyeeS, no further comments from my side.

**Review by registerrier — APPROVED** [review]
Thanks @AtreyeeS ! Looks good to me.

**Pull request merged**